pkg/fqdn/cache.go

```diff
@@ -523,11 +523,31 @@ func (c *DNSCache) upsertReverse(ip string, entry *cacheEntry) {
 	entries[entry.Name] = entry
 }
 
-// removeReverse removes the reference between ip and the name stored in entry.
+// remove removes the reference between ip and the name stored in entry from
+// the DNS cache (both in forward and reverse maps). This assumes the write
+// lock is taken.
+func (c *DNSCache) remove(ip string, entry *cacheEntry) {
```

Comment:
Should the other caller of `removeReverse` in `cleanupOverLimitEntries` at line 303 instead call `remove` as well?

https://github.com/cilium/cilium/blob/bbab9031f05df5506f701f4ef60773a2ca930f21/pkg/fqdn/cache.go#L303